syntax = "proto3";
package messaging.rpc.providers.grpc.proto;


message NodeHeader {
    string uri = 1;
}


message Schedule {
    string id = 1;
    string name = 2;
    string trigger_type = 3;
    int32 type = 4;
    CronSchedule schedule_details = 5;
}


message CronSchedule {
    string second = 1;
    string minute = 2;
    string hour = 3;
    string day = 4;
    string day_of_week = 5; 
    string week = 6;
    string month = 7;
    string year = 8;
    int32 jitter=9;
}


message ClockNode {
    string uuid = 1;
    string uri = 2;
    int32 minute = 3;
    int32 max_schedule_count = 4;
}


// Request response message
message HealthPingRequest {};

message HealthPingResponse {};


message AddScheduleRequest {
    Schedule schedule = 2;
}


message AddScheduleResponse {}


message RemoveScheduleRequest {
    string uuid = 1;
}


message RemoveScheduleResponse {
}


message ReplaceSchedulesRequest {
    repeated Schedule schedules = 1;
}


message ReplaceSchedulesResponse {}


message RegisterClockNodeRequest {
    ClockNode node = 1;
    bool reschedule_on_registration = 2;
}


message RegisterClockNodeResponse {}
